### PR TITLE
Dockerisation update with Dockerfile and requirements.txt

### PR DESCRIPTION
**Pull request by PolarisAstrum**
Added requirements.txt containing python package requirements for pip to install before running run_dx_checker.
Added Dockerfile to allow run_dx_checker to be run using docker.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+FROM python:3.8-slim-bullseye
+
+WORKDIR /
+
+COPY requirements.txt requirements.txt
+
+RUN pip install -r requirements.txt
+
+COPY run_dx_checker.py run_dx_checker.py /
```
Comment (JasonLing95) [inline Dockerfile:11]: duplication

**Review by JasonLing95 — COMMENTED** [review]
how are you reading the environment variables when the docker container is ran

File: Dockerfile (hunk 1 above)
Comment (PolarisAstrum) [reply Dockerfile:11]: What is there a duplication of?

**Comment by PolarisAstrum** [comment]
I didn't make any changes to reading in the environment variables. This takes place in the run_dx_checker.py script, which reads the environment variables from the config.py file and dnanexus_token.py file.

**Comment by JasonLing95** [comment]
could you add a way for docker to read those env variables from its container environment `os.environ['AUTH_TOKEN']` so that when we need to change the env in the future, we could just change it on a config file. 

The command will be `docker run --env-file <the config file path> dx-checker`. And add an [example config](https://github.com/eastgenomics/automated-archiving/blob/main/env.example) too

**Comment by JasonLing95** [comment]
Yup, just in case the API token might expire or we might need to change project-id or workflow-id. It is something you need to add to the python script (run_dx_checker.py) (here's an [example](https://github.com/eastgenomics/automated-archiving/blob/main/main.py))

**Comment by PolarisAstrum** [comment]
Ah, I think this is already a feature. The projectID, workflowID, and AUTH_TOKEN are already specified in the config file. Currently the AUTH_TOKEN is specified in a separate config file named "dnanexus_token.py", while the project and workflow IDs are specified in "config.py". I could move AUTH_TOKEN to "config.py" and remove "dnanexus_token.py" I suppose.

File: Dockerfile (hunk 1 above)
Comment (JasonLing95) [reply Dockerfile:11]: shouldn't this just be `COPY run_dx_checker.py /`

File: run_dx_checker.py [hunk 2]
```diff
@@ -9,8 +9,50 @@
...
-from dnanexus_token import AUTH_TOKEN
+# get environment variables
+AUTH_TOKEN = os.environ['AUTH_TOKEN']
+projectID = os.environ['projectID']
+workflowID = os.environ['workflowID']
+
+inputs = {
+    "0.genomebwaindex_targz": {
+        '$dnanexus_link': {
+            'project': os.environ['genomebwaindex_targz_projectID'],
```
Comment (JasonLing95) [inline run_dx_checker.py:20]: `import os`

File: config.txt [hunk 3]
```diff
@@ -0,0 +1,18 @@
+AUTH_TOKEN=
+
+PROJECT_ID=project-Fy9G62Q4gk5784zB9xPpVFJ1
+WORKFLOW_ID=workflow-Fy8qkxQ433GV4kQv4z8YJ0p0
+
+genomebwaindex_targz_projectID=project-F3zxk7Q4F30Xp8fG69K1Vppj
```
Comment (JasonLing95) [inline config.txt:6]: Capitalize all the env `GENOMEBWAINDEX_TARGZ_PROJECTID`

File: run_dx_checker.py [hunk 4]
```diff
@@ -9,8 +9,50 @@
 
 import dxpy
 
-from config import projectID, workflowID, inputs
-from dnanexus_token import AUTH_TOKEN
+# get environment variables
+AUTH_TOKEN = os.environ['AUTH_TOKEN']
+projectID = os.environ['projectID']
```
Comment (JasonLing95) [inline run_dx_checker.py:14]: Uppercase `projectID` and all env below

**Review by JasonLing95 — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (PolarisAstrum) [reply Dockerfile]: Yep, the command is fixed now

File: run_dx_checker.py (hunk 2 above)
Comment (PolarisAstrum) [reply run_dx_checker.py]: Added

File: config.txt (hunk 3 above)
Comment (PolarisAstrum) [reply config.txt]: All env vars are capitalised now

File: run_dx_checker.py (hunk 4 above)
Comment (PolarisAstrum) [reply run_dx_checker.py]: All env names match in the config.txt and run_dx_checker.py now

**Comment by JasonLing95** [comment]
@PolarisAstrum Could you add a `print()` statement at the end of `main()` function so that we know the script completed running

**Comment by PolarisAstrum** [comment]
The print statement has been added

**Pull request merged**